File: ISBN号码/isbn.py
a=open("./isbn.in","r").read()
b=a[:12] #获取识别号之前的数字
c=0
d=1
for i in b:
    if i!='-': #略过”-“号
        e=int(i)
        c=c+e*d
        d+=1
g=c%11 #计算识别号
if g==10: #如果是十那么就是"X"
    h='X'
else:
    h=str(g)
if h==a[12]:
    jg="Right"
else:
    b=b+h
    jg=b
output_f=open("./isbn.out","w").write(jg) #写入"isbn.out"

File: ISBN号码/isbn.cpp
#include<bits/stdc++.h>
using namespace std;
int main(int argc, char const *argv[]){
	freopen("isbn.in","r",stdin); 
	freopen("isbn.out","w",stdout); 
	char num[10];
	scanf("%c-%c%c%c-%c%c%c%c%c-%c",&num[0],&num[1],&num[2],&num[3],&num[4],&num[5],&num[6],&num[7],&num[8],&num[9]);
	short int cf=0;
	for (short int i = 0; i < 9; ++i)
		cf+=(num[i]-'0')*(i+1);
	if(cf%11==num[9]-'0' ||(cf%11==10 && num[9]=='X'))
		cout<<"Right";
	else{
		if(cf%11!=10)
			cout<<num[0]<<"-"<<num[1]<<num[2]<<num[3]<<"-"<<num[4]<<num[5]<<num[6]<<num[7]<<num[8]<<"-"<<cf%11;
		else
			cout<<num[0]<<"-"<<num[1]<<num[2]<<num[3]<<"-"<<num[4]<<num[5]<<num[6]<<num[7]<<num[8]<<"-X";
	}
	fclose(stdin);
	fclose(stdout);
	return 0;
}
